src/core/options-manager.ts

```diff
@@ -102,18 +104,19 @@ class OptionsManager {
                 return;
             }
 
-            if (!nestedOption.isCollectionItem) {
-                optionValue = getNestedValue(nestedOptionObj, e.fullName.split(".").slice(1));
-            }
+            if (nestedOption.isCollectionItem) { return; }
```

Comment:
This change makes code very tricky. Can we rewrite this method?

Reply:
Gonna refactor this in a different PR